logstash-core/src/main/java/org/logstash/ObjectMappers.java

```diff
@@ -107,21 +122,82 @@ public void serializeWithType(final RubyString value, final JsonGenerator jgen,
                 typeSer.typeId(value, RubyString.class, JsonToken.VALUE_STRING);
             typeSer.writeTypePrefix(jgen, typeId);
             final ByteList bytes = value.getByteList();
-            jgen.writeBinary(bytes.getUnsafeBytes(), 0, bytes.length());
+            final byte[] encoding = ENCODING_PREFIXES.computeIfAbsent(
+                value.getEncoding().getClass(), cls -> cls.getName().getBytes(StandardCharsets.UTF_8)
+            );
+            final ByteArrayOutputStream baos =
+                new ByteArrayOutputStream(1 + bytes.length() + Integer.BYTES + encoding.length) {
+                    @Override
+                    public byte[] toByteArray() {
+                        return buf;
+                    }
+                };
+            baos.write(0xff);
+            final int prefixLen = encoding.length;
+            baos.write((byte) prefixLen >> 24);
```

Comment:
nitpick: using `Ints#toByteArray(int)` here would bring this to the same level of abstraction as the decoder, which uses `Ints#fromBytes(byte, byte, byte, byte)`.

Reply:
@yaauie yea but then we have to allocate another `byte[]` here needlessly. I was just trying to keep the performance impact of adding this low, even though it looks dirty admittedly.